### PR TITLE
docs: add pages to sidebar

### PR DESCRIPTION
**Pull request by alexanderbez**
Add quotas pages to sidebar. ref https://github.com/hashicorp/vault/pull/9342

### CODE REVIEW

File: website/data/api-navigation.js [hunk 1]
```diff
@@ -111,6 +111,7 @@ export default [
       'internal-ui-mounts',
       'key-status',
       'leader',
+      'lease-count-quotas',
```
Comment (mladlow) [inline website/data/api-navigation.js:114]: I think on the actual website these should all be grouped together.

Comment (alexanderbez) [reply website/data/api-navigation.js:114]: `website/data/api-navigation.js` looks to be alphabetically ordered so I was just following convention. I can group these if my assumption is wrong.

Comment (mladlow) [reply website/data/api-navigation.js:114]: I did notice that, but I think the usability of the website should take precedence of the alphabetical ordering of the file? We could rename the file with some kind of prefix if we want to keep the ordering in this file alphabetical, or if there's a linter insisting on alphabetical order in the file.

Comment (alexanderbez) [reply website/data/api-navigation.js:114]: I don't think there is a linter? I'll group them 👍

**Pull request merged**